File: common/src/main/java/org/bread_experts_group/breadlib/task/input/MouseTasks.java
package org.bread_experts_group.breadlib.task.input;

import net.minecraft.client.MouseHandler;
import org.bread_experts_group.breadlib.task.FireSide;
import org.bread_experts_group.breadlib.task.SidedTask;
import org.bread_experts_group.breadlib.task.Task;

public class MouseTasks {
	public static class Scroll extends Task {
		public final double scrollX;
		public final double scrollY;
		public final MouseHandler mouseHandler;

		public Scroll(MouseHandler mouseHandler, double scrollX, double scrollY) {
			this.scrollX = scrollX;
			this.scrollY = scrollY;
			this.mouseHandler = mouseHandler;
		}
	}

	public static class Button extends SidedTask {
		public final int button;
		public final int action;
		public final int modifiers;
		public final MouseHandler mouseHandler;

		public Button(MouseHandler mouseHandler, int button, int action, int modifiers, FireSide side) {
			super(side);
			this.button = button;
			this.action = action;
			this.modifiers = modifiers;
			this.mouseHandler = mouseHandler;
		}
	}
}

File: common/src/main/kotlin/org/bread_experts_group/breadlib/task/input/MouseTasks.kt
package org.bread_experts_group.breadlib.task.input

import net.minecraft.client.MouseHandler
import org.bread_experts_group.breadlib.task.FireSide
import org.bread_experts_group.breadlib.task.SidedTask
import org.bread_experts_group.breadlib.task.Task

class MouseTasks {
	class Scroll(val mouseHandler: MouseHandler, val scrollX: Double, val scrollY: Double) : Task()

	class Button(
		val mouseHandler: MouseHandler,
		val button: Int,
		val action: Int,
		val modifiers: Int,
		side: FireSide
	) : SidedTask(side)
}